layout/brandeskopf.go

```diff
@@ -158,12 +164,43 @@ func horizontalCompaction(g LayeredGraph, root map[uint64]uint64, align map[uint
 		}
 	}
 
+	// class offsets
+	layers := g.Layers()
+	for i := 0; i < len(layers); i++ {
+		layer := layers[i]
+		vfirst := layer[0]
+		if sink[vfirst] == vfirst {
+			if shift[sink[vfirst]] == math.MaxInt {
```

Comment:
hm, dangerous using max ints for overflows, nils may be better.

but I see brandeskopf is already using them, so might be ok then